lib/game.rb

```diff
@@ -0,0 +1,64 @@
+class Game
+  attr_accessor :board, :player_1, :player_2, :cells
+
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [6,4,2]
+  ]
+
+  def initialize(player_1 = Player::Human.new("X"), player_2 = Player::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    board.turn_count.even? ? player_1 : player_2
+  end
+
+  def over?
+    won? || draw?
+  end
+
+  def won?
+    WIN_COMBINATIONS.detect do |combo|
+      combo.all? { |position| board.cells[position] == player_1.token } ||
+      combo.all? { |position| board.cells[position] == player_2.token }
+    end
+  end
+
+  def draw?
+    !won? && board.full?
+  end
+
+  def winner
+    board.cells[won?.first] if won?
```

Comment:
Yep!

On Mon, Apr 18, 2016 at 12:00 AM Shmully Wolfson <notifications@github.com>
wrote:

> In lib/game.rb
> <https://github.com/learn-co-students/ttt-with-ai-project-v-000/pull/126#discussion_r60003940>
> :
>
> > +    won? || draw?
> > +  end
> > +
> > +  def won?
> > +    WIN_COMBINATIONS.detect do |combo|
> > +      combo.all? { |position| board.cells[position] == player_1.token } ||
> > +      combo.all? { |position| board.cells[position] == player_2.token }
> > +    end
> > +  end
> > +
> > +  def draw?
> > +    !won? && board.full?
> > +  end
> > +
> > +  def winner
> > +    board.cells[won?.first] if won?
>
> like this?
>
>   def winner
>     if won = won?
>       board.cells[won.first]
>     end
>   end
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/learn-co-students/ttt-with-ai-project-v-000/pull/126/files/c22b492d245a46964e3e8f421881ed57dcb0da23#r60003940>
>